### PR TITLE
removed confusing logging in get memory handler

### LINKED ISSUE #182: 'what is your favorite yellow object' fails
## Type of Issue

Select the type of issue:
- [x] Bug report (to report a bug)
- [ ] Feature request (to request an additional feature)
- [ ] Tracker (I am just using this as a tracker)
- [ ] Refactor request
- [ ] Documentation Ask

## Description

Error log:
```
2021-02-21 23:40:00,317 [loco_mc_agent.py:263 - controller_step() INFO]: Incoming chat: ['9ba1da83cd35e019e48460cae64542e4dcf70652582aa16d4da025a04ffe4794:a2c15041cbb54f51a35357cf5f2f16a0' -> what is your favorite yellow object]
2021-02-21 23:40:01,564 [dialogue_manager.py:87 - step() INFO]: Dialogue stack pre-run_model: []
2021-02-21 23:40:01,565 [nsp_dialogue_manager.py:276 - get_logical_form() INFO]: Querying the semantic parsing model
2021-02-21 23:40:01,962 [nsp_dialogue_manager.py:293 - get_logical_form() INFO]: chat before lemmatization "what is your favorite yellow object"
2021-02-21 23:40:01,979 [nsp_dialogue_manager.py:296 - get_logical_form() INFO]: chat after lemmatization "what be -PRON- favorite yellow object"
2021-02-21 23:40:01,979 [nsp_dialogue_manager.py:300 - get_logical_form() INFO]: ttad pre-coref "what be -PRON- favorite yellow object" -> {'dialogue_type': 'GET_MEMORY', 'filters': {'triples': [{'pred_text': 'prefers', 'subj_text': '_self'}, {'pred_text': 'has_colour', 'obj_text': 'yellow'}], 'output': {'attribute': 'NAME'}}}
2021-02-21 23:40:01,979 [nsp_dialogue_manager.py:310 - get_logical_form() INFO]: logical form before grammar update "{'dialogue_type': 'GET_MEMORY', 'filters': {'triples': [{'pred_text': 'prefers', 'subj_text': '_self'}, {'pred_text': 'has_colour', 'obj_text': 'yellow'}], 'output': {'attribute': 'NAME'}}}
2021-02-21 23:40:01,979 [nsp_dialogue_manager.py:311 - get_logical_form() INFO]: logical form after grammar fix "{'dialogue_type': 'GET_MEMORY', 'filters': {'triples': [{'pred_text': 'prefers', 'subj_text': '_self'}, {'pred_text': 'has_colour', 'obj_text': 'yellow'}], 'output': {'attribute': 'NAME'}}}"
2021-02-21 23:40:01,980 [nsp_dialogue_manager.py:168 - handle_logical_form() INFO]: logical form post-coref "0f74effcf3087a5df1edeeb5cad7016591c8128bfe54299a28f8127fcaf5e258:b7ca6ea24f934576abc4e2f25549a67b" -> {'dialogue_type': 'GET_MEMORY', 'filters': {'triples': [{'pred_text': 'prefers', 'subj_text': '_self'}, {'pred_text': 'has_colour', 'obj_text': 'yellow'}], 'output': {'attribute': 'NAME'}}}
2021-02-21 23:40:01,982 [nsp_dialogue_manager.py:184 - handle_logical_form() INFO]: this model out: {'dialogue_type': 'GET_MEMORY', 'filters': {'triples': [{'pred_text': 'prefers', 'subj_text': '_self'}, {'pred_text': 'has_colour', 'obj_text': 'yellow'}], 'output': {'attribute': 'NAME'}}}
2021-02-21 23:40:01,984 [get_memory_handler.py:172 - do_answer() ERROR]: list index out of range
Traceback (most recent call last):
  File "/private/home/yuxuans/Workspace/tmp/droidlet/craftassist/agent/../../base_agent/dialogue_objects/get_memory_handler.py", line 166, in do_answer
    return str(vals[0]), None
IndexError: list index out of range
2021-02-21 23:40:01,986 [craftassist_agent.py:242 - send_chat() INFO]: Sending chat: I don't understand what you're asking
```

### PR DESCRIPTION
**Pull request by snyxan**
# Description

The exception logging is confusing, so removed it.

Fixes #182 #176 

## Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

# Testing

Tested locally

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [ ] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have added relevant collaborators to review the PR before merge.

### CODE REVIEW

File: base_agent/dialogue_objects/get_memory_handler.py [hunk 1]
```diff
@@ -169,7 +169,6 @@ def do_answer(self, mems: Sequence[Any], vals: Sequence[Any]) -> Tuple[Optional[
             else:
                 raise ValueError("Bad answer_type={}".format(output_type))
         except Exception as e:
-            logging.exception(e)
```
Comment (anuragprat1k) [inline base_agent/dialogue_objects/get_memory_handler.py:172]: I think it is useful to log this exception. Just prettify it perhaps?

**Review by anuragprat1k — CHANGES_REQUESTED** [review]

**Comment by anuragprat1k** [comment]
Generally, I think we should be logging exceptions since we don't throw them anyway. It is a useful debugging tool. Without it there's no easy way to find out what might be causing an exception. For example in this case, without the trace it would be hard to debug why `do_answer` might be failing.

**Comment by snyxan** [comment]
> Generally, I think we should be logging exceptions since we don't throw them anyway. It is a useful debugging tool. Without it there's no easy way to find out what might be causing an exception. For example in this case, without the trace it would be hard to debug why `do_answer` might be failing.

Added exception logging back and caught IndexError separately. I believe Arthur deliberately throws index errors and let them caught by exception handler. In this way, agent can respond with 'I don't understand what you're asking'.

Since we want: 1. use IndexError as a way to respond with 'IDK', 2. still be able to catch&analyze other unpredictable errors, using two catching here to distinguish them.

File: base_agent/dialogue_objects/get_memory_handler.py [hunk 2]
```diff
@@ -168,9 +168,10 @@ def do_answer(self, mems: Sequence[Any], vals: Sequence[Any]) -> Tuple[Optional[
                 return self.handle_exists(mems)
             else:
                 raise ValueError("Bad answer_type={}".format(output_type))
+        except IndexError: # index error indicates no answer available
```
Comment (anuragprat1k) [inline base_agent/dialogue_objects/get_memory_handler.py:171]: Can you also add a logging statement for the `IndexError` here? We shouldn't be catching errors silently. Adding a logging statement here will help disambiguate the `ErrorWithResponse` thrown here from all the other places that `ErrorWithResponse` could arise from.

**Review by anuragprat1k — APPROVED** [review]
Looks good except one comment that I've left inline.

**Pull request merged**